02_bubble/marchitect/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"math/big"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	// n is the number of items in the int array
+	n := len(s)
+	// maybe necessary
+	if n < 2 {
+		return s
+	}
+	// set swapped to true
+	swapped := true
+	// in Go, use keyword for to act as a while loop
+	for swapped {
+		// set swapped to false
+		swapped = false
+		// iterate through all of the elements in our list
+		for i := 1; i < n; i++ {
+			// if the current element is greater than the next element, swap them
+			if s[i-1] > s[i] {
+				// swap values - Go tuple assignment
+				s[i], s[i-1] = s[i-1], s[i]
+				// set swap back to true
+				swapped = true
+			}
+		}
+	}
+	return s
```

Comment:
The lab asks for returning a sorted copy of int slice s. Returning s is not a copy.
You can make a copy by making a slice then copy:
```
copy := make([]int, len(s))
copy(copy, s)
```

Reply:
Thank you for the feedback, will improve the code accordingly. And yes I aim to spare some time to review more PR too

Reply:
> You will also need to review a colleague's PR.

Kindly please help me check if the updated commit is good to fix the issue :) Thank you David.

Reply:
Nicely done. Looks good to me.